tuf/src/metadata.rs

```diff
@@ -66,7 +66,6 @@ static PATH_ILLEGAL_STRINGS: &[&str] = &[
     "\"",
     "|",
     "?",
-    "*",
```

Comment:
It'd be helpful if we also had a test here so we don't accidentally break this again.